### PR TITLE
add volume lifecycle mode for 1.16

### PR DESCRIPTION
**Pull request by aramase**
fixes https://github.com/deislabs/secrets-store-csi-driver/issues/155

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,7 +138,8 @@ kubectl apply -f manifest.yml
 
 ```bash
 kubectl apply -f deploy/rbac-secretproviderclass.yaml # update the namespace of the secrets-store-csi-driver ServiceAccount
-kubectl apply -f deploy/csidriver.yaml
+kubectl apply -f deploy/csidriver.yaml # deploy this yaml for kubernetes version 1.15 only
+kubectl apply -f deploy/csidriver-1.16.yaml # deploy this yaml for kubernetes version 1.16+
```
Comment (ritazh) [inline README.md]: should we make the 1.16+ one as the default? and comment out the <1.16 one as optional if they are on <1.16?

Comment (aramase) [reply README.md]: @ritazh Are you suggesting we add the `volumeLifecycleMode` to `csidriver.yaml` and create a different one for 1.15 only (`csidriver-1.15.yaml`)?

Comment (ritazh) [reply README.md]: yes

Comment (aramase) [reply README.md]: done!

**Pull request merged**